### ISSUE TITLE
Review IDAES dependencies 

### ISSUE THREAD

**Issue by dangunter** [IDAES/idaes-pse#133]
Review pinned and unpinned Python dependencies in `setup.py`

**Comment by Andresj89** [comment]
Hi guys, I just tried to build the following `pip install -r requirements-dev.txt` and I got this message by the end of the installation process:
```
Successfully built PyUtilib Pyomo coverage sphinx-argparse jupyter-latex-envs jupyter-nbextensions-configurator wrapt
Installing collected packages: PyUtilib, Pyomo, coverage, mccabe, pycodestyle, pyflakes, flake8, jupyter-latex-envs, jupyter-contrib-core, jupyter-highlight-selected-word, jupyter-nbextensions-configurator, jupyter-contrib-nbextensions, wrapt, lazy-object-proxy, astroid, isort, pylint, pytest-cov, python-coveralls, snowballstemmer, sphinx-rtd-theme, pockets, sphinxcontrib-napoleon, sphinx-argparse, idaes-pse   
  Attempting uninstall: PyUtilib
    Found existing installation: PyUtilib 6.0.0
    Uninstalling PyUtilib-6.0.0:
      Successfully uninstalled PyUtilib-6.0.0
  Attempting uninstall: Pyomo
    Found existing installation: Pyomo 5.7.2
    Uninstalling Pyomo-5.7.2:
      Successfully uninstalled Pyomo-5.7.2
  Attempting uninstall: snowballstemmer
    Found existing installation: snowballstemmer 2.1.0
    Uninstalling snowballstemmer-2.1.0:
      Successfully uninstalled snowballstemmer-2.1.0
  Attempting uninstall: idaes-pse
    Found existing installation: idaes-pse 1.9.0.dev0
    Uninstalling idaes-pse-1.9.0.dev0:
      Successfully uninstalled idaes-pse-1.9.0.dev0
  Running setup.py develop for idaes-pse
ERROR: After October 2020 you may experience errors when installing or updating packages. This is because pip will change the way that it 
resolves dependency conflicts.

We recommend you use --use-feature=2020-resolver to test your packages with the new resolver before it becomes the default.

pytest-cov 2.11.1 requires coverage>=5.2.1, but you'll have coverage 4.5.4 which is incompatible.
Successfully installed PyUtilib-6.0.1.dev0 Pyomo-6.0.dev0 astroid-2.5.3 coverage-4.5.4 flake8-3.9.1 idaes-pse isort-5.8.0 jupyter-contrib-core-0.3.3 jupyter-contrib-nbextensions-0.5.1 jupyter-highlight-selected-word-0.2.0 jupyter-latex-envs-1.4.6 jupyter-nbextensions-configurator-0.4.1 lazy-object-proxy-1.6.0 mccabe-0.6.1 pockets-0.9.1 pycodestyle-2.7.0 pyflakes-2.3.1 pylint-2.7.4 pytest-cov-2.11.1 python-coveralls-2.9.3 snowballstemmer-1.2.1 sphinx-argparse-0.2.5 sphinx-rtd-theme-0.5.2 sphinxcontrib-napoleon-0.7 wrapt-1.12.1
```
Any suggestions as to how to workaround this?

**Comment by ksbeattie** [comment]
@Andresj89, I can't seem to reproduce this.  Can you provide more info how to created the env here, other `pip install` or `pip upgrade` commands run in this env before you got this message?  Perhaps the current output of a `pip freeze`?

Given the above, it looks like you ran that `pip install -r requirements-dev.txt` in an env where you had already installed `idaes-pse` and upgraded (among other things) `pytest-cov` to `2.11.1` which is beyond what you'd get if you installed in a new env (because `coverage` is fixed to `4.5.4` in setup.py).  Is that right?

I think the simplest thing would be to start in a new conda env and try a clean install again.

All this aside, we need to revisit if still need to fix `coverage` to `4.5.4` in `setup.py`.

**Comment by andrewlee94** [comment]
@ksbeattie Coverage was only fixed to `4.5.4` because of `coveralls` which we are no longer using.

**Comment by Andresj89** [comment]
@ksbeattie before installing the requirements I did `pip install sphinx sphinx-autobuild` and then `pip install rstcheck`

**Comment by ksbeattie** [comment]
This is getting addressed in PR #292
